app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @review = Review.new(comment_params)
+    @review.user = current_user
+    if @review.save
+      flash[:success] = "Comment successful."
+      redirect_to shop_product_path(@product.shop, @product)
+    end
+
+    # @star_rating = @product.star_rating.new :value => params[:value]
+    # if @star_rating.save
+    #   flash[:success] = "Rating successful."
+    #   redirect_to shop_product_path(@product.shop, @product)
+    # else
+    #   flash[:success] = "Something went wrong."
+    #   redirect_to shop_product_path(@product.shop, @product)
+    # end
+  end
+
+  def edit
+	end
+
+	def update
+		if @review.update(update_comment_params)
+			flash[:success] = "Your comment was successfully updated"
```

Comment:
You REALLY need to fix the indentation issues @LevinGeorge :P Settings for tab are 2 spaces.